near/omni-token/src/migrate.rs

```diff
@@ -0,0 +1,79 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_contract_standards::fungible_token::{FungibleToken, metadata::FungibleTokenMetadata};
+use near_sdk::{AccountId, Gas, NearToken, collections::LazyOption, env, near, require, store::Lazy};
+use crate::{OmniToken, OmniTokenExt, WITHDRAW_RELAYER_ADDRESS, omni_ft::UpgradeAndMigrate};
+use near_sdk::serde_json::json;
+
+const CURRENT_STATE_VERSION: u32 = 3;
+const OUTER_UPGRADE_GAS: Gas = Gas::from_tgas(15);
+const NO_DEPOSIT: NearToken = NearToken::from_yoctonear(0);
+const STATE_KEY: &[u8] = b"STATE";
+const OWNABLE_KEY: &[u8] = b"OWNABLE";
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct NearIntentsState {
+    pub token: FungibleToken,
+    pub metadata: Lazy<FungibleTokenMetadata>,
+}
+
+#[near]
+impl OmniToken {
+    #[private]
+    #[init(ignore_state)]
+    #[allow(unused_variables)]
+    pub fn migrate(controller: AccountId, withdraw_relayer: Option<AccountId>) -> Self {
```

Comment:
The migration function is private, so only the self account can call it.
We have to align on the migration flow before releasing this.

Reply:
It's built on top of global contracts PR.
Migration will be done with full access key so private is fine.